### ISSUE TITLE
Change Requests | Capture & Expose Status Change History

### ISSUE THREAD

**Issue by sethmcknight** [SeedCompany/cord-api-v3#2273]
Ok so one step at a time. I think we need this new schema:

type Mutation {
  transitionProject(input: ProjectTransitionInput!): ProjectTransitionOutput!
}
input ProjectTransitionInput {
  id: ID!
  changeset: ID
  step: ProjectStep!
  comment: String
}
type ProjectTransitionOutput {
  project: Project!
}

type SecuredProjectStep {
  history: [ProjectStepChange!]!
}
type ProjectStepChange {
  step: ProjectStep!
  by: User!
  comment: String
}
Does transitionProject() or changeProjectStep() make more sense?
These should fully work with changesets
Maybe we deprecate updateProject.step for this new mutation as well?
Once this is done we can talk about integrating changeset status and project step workflows.

**Comment by sethmcknight** [comment]
@zkhin we pulled this work out of https://github.com/SeedCompany/cord-api-v3/issues/2084 to work as a separate issue.

**Comment by zkhin** [comment]
@sethmcknight Thank you!

**Comment by rdonigian** [comment]
Hold for historical reference post db migration.
